drizzle-orm/package.json

```diff
@@ -177,6 +177,7 @@
 		"@vercel/postgres": "^0.8.0",
 		"@xata.io/client": "^0.29.3",
 		"better-sqlite3": "^8.4.0",
+		"bson": "^6.8.0",
```

Comment:
I'm curious about how we want to represent the data here. Do we expect users to be dealing with BSON data in a manner like this, or are we expecting them to convert it to JSON anyways? Especially considering the to-driver stuff is using `JSON.stringify`

Reply:
From my perspective, the ideal scenario is that users deal only with JSON strings and let the form deal with the translation. That would make things more easy for users. 

Something like:

```ts
db.singlestoreTable({
  ...
   bson_column: bson().notnull().default({"key": "value"})
   ...
 }) 
 ```
 
Currently there is a bug on the translation of literals into BSON on the engine side, so this will only be possible once the bug on the engine is fixed. For now the default value has to be converted to BSON object and then to hexa string
 
 More info about the bug on slack: https://memsql.slack.com/archives/C02G51BAJ/p1724069568691219?thread_ts=1724061645.083329&cid=C02G51BAJ
 
I also found out that you can cast json literals on select and insert statements, but not on create table statements. Because of that, we can keep the `jsonString:>BSON` on the `mapToDriver` return value, since the drizzle-orm package only deals with insert and select operations, and the option above on drizzle-kit, that deals with scheam modifications (create table and create/alter column)

Reply:
Yeah, that all makes sense from the engine side. I guess I was more curious about the JS side of when we get the data out from the engine. Do we expect users to use BSON directly in TypeScript or should we also convert it back to JSON?